File: src/main/java/ru/ezhov/rocket/action/RocketActionSettingsRepository.java
package ru.ezhov.rocket.action;

import ru.ezhov.rocket.action.api.RocketActionSettings;

import java.util.List;

public interface RocketActionSettingsRepository {
    List<RocketActionSettings> actions() throws RocketActionSettingsRepositoryException;

    void save(List<RocketActionSettings> settings) throws RocketActionSettingsRepositoryException;
}

File: src/main/kotlin/ru/ezhov/rocket/action/RocketActionSettingsRepository.kt
package ru.ezhov.rocket.action

import ru.ezhov.rocket.action.api.RocketActionSettings

interface RocketActionSettingsRepository {
    @Throws(RocketActionSettingsRepositoryException::class)
    fun actions(): List<RocketActionSettings>

    @Throws(RocketActionSettingsRepositoryException::class)
    fun save(settings: List<RocketActionSettings>)
}